mcs/class/System/Test/System.Diagnostics/ProcessTest.cs

```diff
@@ -1035,6 +1036,18 @@ public void DisposeWithDisposedStreams ()
 		[NUnit.Framework.Category ("MobileNotWorking")]
 		public void StandardInputWrite ()
 		{
+			// Because we have no NotWorking category for AIX, use
+			// reflection to access the internal System.Platform
+			// as recommended. (XXX: RuntimeInformation instead?)
+			var platform = Assembly.Load("System")
+				?.GetType("System.Platform");
+			if ((bool)platform?.GetMethod("get_IsAix")?.Invoke(platform, null) == true
+				|| (bool)platform?.GetMethod("get_IsIBMi")?.Invoke(platform, null) == true)
```

Comment:
I think we should add AIX/IBMi to RuntimeInformation instead of doing this reflection magic.

```diff
diff --git a/mcs/class/corlib/System.Runtime.InteropServices.RuntimeInformation/RuntimeInformation.cs b/mcs/class/corlib/System.Runtime.InteropServices.RuntimeInformation/RuntimeInformation.cs
index 7b11484ead0..494e1f5d700 100644
--- a/mcs/class/corlib/System.Runtime.InteropServices.RuntimeInformation/RuntimeInformation.cs
+++ b/mcs/class/corlib/System.Runtime.InteropServices.RuntimeInformation/RuntimeInformation.cs
@@ -62,6 +62,13 @@ namespace System.Runtime.InteropServices
                        case PlatformID.Unix:
                                return osPlatform == OSPlatform.Linux;
                        default:
+                               if (Platform.IsFreeBSD)
+                                       return osPlatform == OSPlatform.Create ("FREEBSD");
+                               if (Platform.IsIBMi)
+                                       return osPlatform == OSPlatform.Create ("IBMI");
+                               if (Platform.IsAix)
+                                       return osPlatform == OSPlatform.Create ("AIX");
+
                                return false;
                        }
 #endif
```

Reply:
Ugh, Platform is in System, so you still need to cross the reflection boundary for it. How about a uname based approach for Unix, if possible; or DllMap based? I'm thinking another icall....

Reply:
sorry, forgot about that 😭

how about we move `Platform` into corlib? we have InternalsVisibleTo from corlib -> System so this should work.

Reply:
Sure, that works too. Would you want to do that, or me?

Reply:
To sum up what we discussed on Gitter:

- Adding a new icall that gets the OS as used by the runtime for dllmap, similar to the existing RuntimeArchitecture.
- Use that for the existing IsOSPlatform implementation
- Cache both values in a static field so we don't need to the icall every time